File: src/main/java/net/hectus/neobb/turn/default_game/mob/TEvoker.java
package net.hectus.neobb.turn.default_game.mob;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.CounterFilter;
import net.hectus.neobb.turn.default_game.attributes.clazz.SupernaturalClazz;
import net.hectus.neobb.turn.default_game.attributes.function.CounterbuffFunction;
import org.bukkit.DyeColor;
import org.bukkit.entity.Evoker;

import java.util.List;

public class TEvoker extends MobTurn<Evoker> implements CounterbuffFunction, SupernaturalClazz {
    public TEvoker(NeoPlayer player) { super(player); }
    public TEvoker(Evoker data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Luck(Buff.BuffTarget.YOU, 20), new Buff.ExtraTurn());
    }

    @Override
    public List<CounterFilter> counters() {
        return List.of(CounterFilter.of(turn -> turn instanceof TSheep t && t.data().getColor() == DyeColor.BLUE, "blue-sheep"));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/mob/TEvoker.kt
package net.hectus.neobb.modes.turn.default_game.mob

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.modes.turn.default_game.CounterFilter
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.SupernaturalClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.CounterbuffFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.DyeColor
import org.bukkit.entity.Evoker

class TEvoker(data: Evoker?, cord: Cord?, player: NeoPlayer?) : MobTurn<Evoker>(data, cord, player), CounterbuffFunction, SupernaturalClazz {
    override val cost: Int = 4

    override fun buffs(): List<Buff<*>> {
        return listOf(Luck(20), ExtraTurn())
    }

    override fun counters(): List<CounterFilter> {
        return listOf(CounterFilter.of("blue-sheep") { turn -> turn is TSheep && turn.data!!.color == DyeColor.BLUE })
    }
}
